Shell/AST.h

```diff
@@ -122,7 +122,7 @@ struct PathRedirection : public Redirection {
 };
 
 struct FdRedirection : public Redirection
-    , public Rewiring {
+    , virtual Rewiring {
```

Comment:
Wait, what's even going on here..
`Redirection` and `Rewiring` are both `RefCounted`. That's no good, that gives us two reference counts in the same object. That needs to be fixed.

Reply:
That's what I'm trying right now.
Not sure what I was doing when I came up with that idea.